.github/workflows/docker-hub-push.yml

```diff
@@ -0,0 +1,108 @@
+# focuses on pushing pycytominer docker images to docker hub
+name: docker-push
+
+env:
+  LANG: "en_US.utf-8"
+  LC_ALL: "en_US.utf-8"
+  PYTHONIOENCODING: "UTF-8"
+  TARGET_PYTHON_VERSION: "3.9"
+
+on:
+  schedule:
+    # runs every Wednesday at 7 AM UTC
+    - cron: "0 7 * * 3"
+  # trigger for pushes to main
+  push:
+    branches: [main]
+  # trigger for published releases (distinct from drafts)
+  release:
+    types:
+      - published
+
+jobs:
+  docker-push:
+    runs-on: ubuntu-latest
+    steps:
+      # checks out selected files for docker image build testing
+      - name: Checkout selected files
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            build/docker/Dockerfile
+          sparse-checkout-cone-mode: false
+      - name: Download sdist artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: sdist
+      # Gather and prepare the dynamic version and other vars for use with work below.
+      # For example, 1.1.0.post6.dev0+bc093ef becomes 1.1.0.post6.dev0_bc093ef
+      - name: Gather dynamic version and set env var
+        id: env-vars
+        run: >-
+          VERSION=$(
+            find ./*.tar.gz |
+            sed 's/pycytominer-\(.*\)\.tar\.gz/\1/' |
+            tr '+' '_'
+          ) &&
+          echo "VERSION=$VERSION" >> "$GITHUB_OUTPUT" &&
+          echo "SDIST_FILENAME=$(find ./*.tar.gz)" >> "$GITHUB_OUTPUT" &&
+          echo "SDIST_EXTRACTED_NAME=\"pycytominer-$VERSION\"" >> "$GITHUB_OUTPUT" &&
```

Comment:
If we switch to making this part of the ci overall ci workflow, I would prefer we get these vars from `needs.upload.output` instead of using string parsing. String parsing is hard to ... parse ... 🥁 and is a common source of bugs in my view. 

If we get these variables from an appropriate output, this particular step could then be limited to generating a `CLEAN_VERSION` variable (without the `+`) and the `DATE_TAG`.

Reply:
Thanks, agreed on parsing 😆 ! I'll move towards limiting the creation of these vars as you described.